bloom-instance/s3.tf

```diff
@@ -1,3 +1,40 @@
 resource "aws_s3_bucket" "user_upload_bucket" {
   bucket_prefix = "${var.name_prefix}-user-uploads"
 }
+
+resource "aws_s3_bucket_cors_configuration" "user_upload_cors_config" {
+  bucket = aws_s3_bucket.user_upload_bucket.id
+
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["PUT", "GET"]
```

Comment:
Why do we need to allow cross-origin PUTs?  Aren't all PUTs done on the server side?

Reply:
We do need to allow cross-origin PUTs. Without this change, PUT requests fail with this error:

Access to XMLHttpRequest at '[...]&x-id=PutObject' from origin 'http://localhost:3001' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

Reply:
This warrants additional investigation.  I can't imagine a scenario where there would be a localhost origin set on a request originating from server-side code.  Are you sure there isn't something making a PUT from the client side?  Because that's all kinds of bad news even for the current Bloom instance if it is